wxGrid: remove trailing newline if only one cell selected

Blake-Madden opened the pull request.

Blake-Madden committed:
wxGrid: remove trailing newline if only one cell selected

vadz reviewed (commented):
Thanks, this looks fine, even though looking at this code again I can't help wondering why do we add the extra new line only to remove it later instead of doing what the inner loop does and only adding it if necessary.

I.e. I'd rather write it like this

```cpp
bool firstRow = true;
for ( ... row ... )
{
    if ( firstRow )
        firstRow = false;
    else
        buf += wxTextFile::GetEOL();

    ... same code as now, maybe renaming "first" to "firstCol", but without += at the end ...
}
```

Wouldn't this be more clear/consistent?

Blake-Madden committed:
Refactor newline logic for cell row copying
This won't include a trailing newline anymore, but I prefer that myself. Any objections?

Blake-Madden commented:
> Thanks, this looks fine, even though looking at this code again I can't help wondering why do we add the extra new line only to remove it later instead of doing what the inner loop does and only adding it if necessary.
> 
> I.e. I'd rather write it like this
> 
> ```c++
> bool firstRow = true;
> for ( ... row ... )
> {
>     if ( firstRow )
>         firstRow = false;
>     else
>         buf += wxTextFile::GetEOL();
> 
>     ... same code as now, maybe renaming "first" to "firstCol", but without += at the end ...
> }
> ```
> 
> Wouldn't this be more clear/consistent?

Got it. I re-implemented as requested.

vadz commented:
Looks good, thanks for the update! I will merge it soon.
